frontend/src/pages/matching-service/WaitForMatchingPage.tsx

```diff
@@ -43,40 +44,49 @@ export default function WaitForMatchingPage() {
   //   return () => clearInterval(intervalId);
   // }, []);
 
-  const checkMatchingState = () => {
+  const checkMatchingState = async () => {
     console.log("checking matching state");
-    if(location.pathname !== pathname) {
-      cancelMatching();
-      console.log("matching cancelled due to leaving page");
-      return;
+    
+    // Check if the pathname has changed
+    if (location.pathname !== pathname) {
+        cancelMatching();
+        console.log("matching cancelled due to leaving page");
+        return;
     }
-    sendCheckMatchingStateRequest(auth.id).then(
-      response => {
-        if(response.status === 200) {
-          console.log("match found!");
-          cancelMatching(false);
-          const roomId = response.roomId
-          navigate(`../collaboration/${roomId}`);
-        } else if (response.status === 202) {
-          //Do nothing
-          console.log("matching...");
+
+    // Send request to check matching state
+    const response = await sendCheckMatchingStateRequest(auth.id);
+    
+    if (response.status === 200) {
+        console.log('2 users are matched')
+        // Check if the required information is in collaboration-service, before navigating to collab page
+        const response = await isUserInCollabStore(auth.id);
```

Comment:
i think this logic should be in matching not collab so if a user is already collabing, the user should not be matched in the first place

Reply:
thats why i mentioned collab service sending matching service the updated list of users collabing so that when first starting to match users, matching service will compare the user against that list to avoid this scenario